h_malloc.c

```diff
@@ -461,19 +484,43 @@ static void set_slab_canary_value(UNUSED struct slab_metadata *metadata, UNUSED
         0x00ffffffffffffffUL;
 
     metadata->canary_value = get_random_u64(rng) & canary_mask;
+#ifdef HAS_ARM_MTE
+    if (unlikely(metadata->canary_value == 0)) {
+        metadata->canary_value = 0x100;
```

Comment:
Why is the canary set to `0x100` is MTE is available?

Reply:
See https://github.com/GrapheneOS/hardened_malloc/pull/225/commits/597d0c3064b75db103d154071c7f3c1cba708d44

Canaries have to be re-enabled if MTE is disabled at runtime. Allocations that were made when MTE was on have to be deallocate-able after it's turned off.

Reply:
Ok, but why `0x100`?

Reply:
Added a comment https://github.com/GrapheneOS/hardened_malloc/pull/225/commits/ca8e9d7958402336db4247f9c41e05d46b231ddd